pkg/scheduler/objects/preemption_queue_test.go

```diff
@@ -53,101 +53,156 @@ type res struct {
+	assertMessageStruct2 := assertMessage{assertMessage2, assertMessage2, assertMessage1, assertMessage2}
+	assertMessageStruct3 := assertMessage{assertMessage3, assertMessage3, assertMessage1, assertMessage1}
+	assertMessageStruct4 := assertMessage{assertMessage3, assertMessage3, assertMessage4, assertMessage1}
+
+	var tests = []struct {
+		testName  string
+		resources res
+	}{
+		{
+			// no guaranteed and no usage. so nothing to preempt
+			testName: "NoGuaranteedNoUsage",
+			resources: res{
+				guaranteed:     resource{nil, nil, nil, nil},
+				allocated:      allocated1,
+				remaining:      preemptable1,
+				assertMessages: assertMessageStruct1,
+			},
+		},
+		{
+			// yes guaranteed and no usage. so nothing to preempt
+			testName: "YesGuaranteedNoUsage",
+			resources: res{
+				guaranteed:     guaranteed1,
+				allocated:      resource{nil, nil, nil, nil},
+				remaining:      preemptable1,
+				assertMessages: assertMessageStruct1,
+			},
+		},
+		{
+			// add usage to parent + root: use all guaranteed at parent level
+			// add usage to child2: use double than guaranteed
+			// usage has exceeded guaranteed for some queues, preemptable resources should be as expected
+			testName: "GuaranteedSetUsageExceeded",
+			resources: res{
+				guaranteed:     guaranteed2,
+				allocated:      allocated2,
+				remaining:      preemptable5,
+				assertMessages: assertMessageStruct2,
+			},
+		},
+		{
+			// add usage to parent + root: use all guaranteed at parent level
+			// add usage to child2: use double than guaranteed
+			// usage has exceeded guaranteed for some queues, preemptable resources should be as expected
+			testName: "GuaranteedSetUsageExceeded",
+			resources: res{
+				guaranteed:     guaranteed1,
+				allocated:      allocated2,
+				remaining:      preemptable2,
+				assertMessages: assertMessageStruct2,
+			},
+		},
+		{
+			// add usage for all: use exactly guaranteed at parent and child level
+			// parent guarantee used for one type child guarantee used for second type
+			testName: "GuaranteedSetExactUsage",
+			resources: res{
+				guaranteed:     guaranteed1,
+				allocated:      allocated3,
+				remaining:      preemptable3,
+				assertMessages: assertMessageStruct3,
+			},
+		},
+		{
+			// add usage for root + parent: use exactly guaranteed at parent and child level
+			// add usage to child1: use double than guaranteed
+			// parent guarantee used for one type child guarantee used for second type
+			testName: "GuaranteedSetMixedUsage",
+			resources: res{
+				guaranteed:     guaranteed1,
+				allocated:      allocated4,
+				remaining:      preemptable4,
+				assertMessages: assertMessageStruct4,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.testName, func(t *testing.T) {
+			rootQ, parentQ, childQ1, childQ2 := setupQueueStructure(t)
+			assertPreemptable(t, rootQ, parentQ, childQ1, childQ2, tt.resources)
```

Comment:
Don't extract assertPreemptable() here, but leave the logic inline. This makes it *much* easier to see where failures occur, as stack traces will show this instead of the generic line in assertPreemptable().

Reply:
Agree, Done.